### PR TITLE
breaking: deprecate constants from defaults

### PR DESCRIPTION
**Pull request by metrizable**
*Issue #, if available:* #1465 

*Description of changes:*

changes include:

* remove version constants from **/defaults.py from most of the frameworks
* include additional test fixtures
* small cli v1 modifications
* several TODO comments

*Testing done:*

unit:
```
% tox --parallel 3 tests/unit
✔ OK black-format in 10.368 seconds
✔ OK flake8 in 25.446 seconds
✔ OK twine in 15.386 seconds
✔ OK pylint in 39.766 seconds
✔ OK doc8 in 23.329 seconds
✔ OK py36 in 50.628 seconds
✔ OK sphinx in 1 minute, 30.923 seconds
✔ OK py37 in 48.687 seconds
✔ OK py27 in 2 minutes, 9.817 seconds
```

integ:

```
% export IGNORE_COVERAGE="#"
% tox -q -e py37 -- -n 3 -rfE --disable-warnings -s -vv tests/integ/test_airflow_config.py -k airflow_config_uploads_data_source_to_s3 -k "tf or xgboost"
...
tests/integ/test_airflow_config.py::test_tf_airflow_config_uploads_data_source_to_s3[2.2.0]
[gw0] PASSED tests/integ/test_airflow_config.py::test_tf_airflow_config_uploads_data_source_to_s3[2.2.0]
...
% tox -q -e py37 -- -n 3 -rfE --disable-warnings -s -vv tests/integ/test_transformer.py::test_transform_tf_kms_network_isolation
...
[gw2] PASSED tests/integ/test_transformer.py::test_transform_tf_kms_network_isolation
```

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to any/all clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/amazon/amazon_estimator.py [hunk 1]
```diff
@@ -611,6 +611,8 @@ def get_image_uri(region_name, repo_name, repo_version=1):
     """Return algorithm image URI for the given AWS region, repository name, and
     repository version
 
+    TODO: consider refactoring version/repo validation into a module for all frameworks
+
```
Comment (laurenyu) [inline src/sagemaker/amazon/amazon_estimator.py:615]: I'm not sure I buy the usefulness of this comment here. We already have #1464.

File: src/sagemaker/xgboost/defaults.py [hunk 2]
```diff
@@ -15,14 +15,20 @@
 
 XGBOOST_NAME = "xgboost"
 XGBOOST_1P_VERSIONS = ["1", "latest"]
-XGBOOST_VERSION_0_90 = "0.90"
+
+# TODO: evaluate usefulness of these constants. they are defined here and only used here
```
Comment (laurenyu) [inline src/sagemaker/xgboost/defaults.py:19]: they're used in `XGBOOST_SUPPORTED_VERSIONS` which is used in `get_image_uri` iirc

File: tests/conftest.py [hunk 3]
```diff
@@ -44,15 +38,19 @@
 
 NO_T2_REGIONS = ["eu-north-1", "ap-east-1", "me-south-1"]
 
+# TODO: refactor handling of versions, repo, image uris, validations for all frameworks
+TENSORFLOW_LATEST_VERSION = "2.2.0"
+TENSORFLOW_LATEST_1X_VERSION = "1.15.2"
```
Comment (laurenyu) [inline tests/conftest.py:43]: I think we can simplify the TF version stuff here and just use 2.x (like how the other frameworks just test one version for integ tests). I don't think the Python SDK behaves differently in our integ tests between TF versions 1.15.2 and 2.2.0

File: src/sagemaker/amazon/amazon_estimator.py (hunk 1 above)
Comment (metrizable) [reply src/sagemaker/amazon/amazon_estimator.py:615]: will remove

File: tests/conftest.py (hunk 3 above)
Comment (metrizable) [reply tests/conftest.py:43]: sounds good, will simplify

**Comment by metrizable** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: sagemaker-python-sdk-notebook-tests
* Commit ID: 3db0ab214aeb12a3dd8073b74c581ed34a231e7d
* Result: FAILED
* [Build Logs](https://3rpiknuts4.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=7090b335-2b27-462e-bd27-36abc88388e5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: tests/conftest.py [hunk 4]
```diff
@@ -291,17 +285,28 @@ def sklearn_full_version(request):
     return request.config.getoption("--sklearn-full-version")
 
 
-@pytest.fixture(scope="module", params=[TensorFlow._LATEST_1X_VERSION, LATEST_VERSION])
+@pytest.fixture(scope="module")
+def tf_latest_version(request, tf_full_version):
+    return request.param
```
Comment (laurenyu) [inline tests/conftest.py:290]: I think you can scrap this in favor of `tf_full_version`?

Comment (metrizable) [reply tests/conftest.py]: done

**Pull request merged**